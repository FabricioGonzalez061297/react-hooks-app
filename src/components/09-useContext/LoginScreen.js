import React, { useContext } from 'react'
import { UserContext } from './UseContext';

export const LoginScreen = () => {

    const {setUser}= useContext(UserContext);

    return (
        <div>
            <h1>Login Screen</h1>
            <hr></hr>

            <button className="btn btn-primary" 
                onClick={()=>{
                    setUser({
                        id:123,
                        name:'Fabricio'
                    });
                }}
            >
                Login
            </button>
            
        </div>
    )
}
