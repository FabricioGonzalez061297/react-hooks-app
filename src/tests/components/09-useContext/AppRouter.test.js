import { mount } from 'enzyme';
import React from 'react';
import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { UserContext } from '../../../components/09-useContext/UseContext';

describe('Pruebas en <AppRouter/>', () => {

    const user={
        name:'fabricio',
        id:1
    }

    const wrapper= mount(
        <UserContext.Provider value={{user}} >
            <AppRouter/>
        </UserContext.Provider>
    )

    test('deberÃ­a crearce correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    


})
