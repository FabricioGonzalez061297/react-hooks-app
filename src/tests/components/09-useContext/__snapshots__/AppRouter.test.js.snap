// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter/> deber√≠a crearce correctamente 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        Context App
      </a>
      <button
        aria-controls="navbarNavAltMarkup"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarNavAltMarkup"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current="page"
            className="nav-link active active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Inicio
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div
    className="container mt-3"
  >
    <div>
      <h1>
        Home Screen
      </h1>
      <hr />
      <pre>
        {
   "name": "fabricio",
   "id": 1
}
      </pre>
    </div>
  </div>
</div>
`;
